src/coreclr/tools/aot/DependencyGraphViewer/Program.cs

```diff
@@ -143,16 +146,125 @@ struct GraphEvent
         public int Num3;
         public string Str;
     }
+    class DGMLGraphProcessing
+    {
+        ConcurrentQueue<GraphEvent> events = new ConcurrentQueue<GraphEvent>();
+
+        public delegate void OnCompleted(int currFileID);
+        public event OnCompleted Complete;
+
+        public DGMLGraphProcessing(int file)
+        {
+            FileID = file;
+        }
+        public int FileID
+        {
+            get; init;
+        }
+
+        public void FindXML(string argPath)
+        {
+            var fileContent = string.Empty;
+            var filePath = string.Empty;
+            var fileStream = System.IO.Stream.Null;
+
+            using (OpenFileDialog openFileDialog = new OpenFileDialog())
+            {
+                openFileDialog.InitialDirectory = argPath;
+                openFileDialog.Filter = @"All Files|*.*|DGML(*.dgml)|*.dgml|XML(*.xml)|*xml";
+                openFileDialog.FilterIndex = 2;
+                openFileDialog.RestoreDirectory = true;
+
+                if (!argPath.EndsWith(".dgml") && !argPath.EndsWith(".xml"))
+                {
+                    if (openFileDialog.ShowDialog() == DialogResult.OK)
+                    {
+                        //Get the path of specified file
+                        filePath = openFileDialog.FileName;
+                        fileStream = openFileDialog.OpenFile();
+                    }
+                }
+                else
+                {
+                    if (argPath.EndsWith(".dgml")) // all .dgml files in the path need to have .xml added for XmlReader to function
+                    {
+                        argPath += ".xml";
+                    }
+                    fileStream = new FileStream(argPath, FileMode.Open);
```

Comment:
I'm not sure that this will work, so basically a user gives you a .dgml file and then you use FileStream to open a handle on a .dgml.xml that is non-existent in the file system

Reply:
Didn't notice that Sven already commented on this, resolving this one to not have duplication